snap/snapcraft.yaml

```diff
@@ -38,6 +34,17 @@ apps:
     plugs: [network, network-bind]
 
 parts:
+  version:
+    plugin: nil
+    source: snap/local
+    override-pull: |
+      cd $SNAPCRAFT_PROJECT_DIR
+      if [ -f VERSION ]; then
+        PROJECT_VERSION=$(cat VERSION)
+      else
+        PROJECT_VERSION=local-dev
```

Comment:
Do we want this to be consistent with what the the service reports as the version? Currently in this case the service would report `0.0.0`

Reply:
@lenny-intel Thanks, that's a good question. I was trying to standardise the way we do this across the snaps - and edgex-go currently uses "local-dev" as default if the version is not supplied. However, you're absolutely right - the default behaviour in the Docker/native build is to set the version to 0.0.0:

https://github.com/edgexfoundry/device-camera-go/blob/c3d329a14a61f436965600af0655521d0b5b2c7b/Makefile#L10

and it makes sense to use that instead, so that there is consistency across different build methods. Otherwise, if a snap is built without a VERSION file, the version used in the snap filename and version string is "local-dev", but the api/v1/version call will return "0.0.0", so there is an inconsistency there.

So I think this should be changed as you suggest to 0.0.0 - and that we should also change the version used in the edgexfoundry (edgex-go) snap accordingly to use 0.0.0 (although that may have to wait until after code freeze).

@tonyespy do you agree as well?

Reply:
@siggiskulason

Frankly, I still find this approach (i.e. VERSION being an untracked/generated file) to versioning clunky, and would prefer if everything was just tag based, but it's a bit late to be thinking about this now, so +1 for aligning the snap versioning with the Docker versioning.

To be clear, we'll always have a properly set `VERSION` files in all of our mirrors, so we'll see tag-based versions for all of our snaps pushed to the store, correct?

Reply:
@tonyespy - yes, the edgexfoundry repositories do not come with a VERSION file. Our snap build pipeline creates a VERSION file containing the latest tag, before building the snap. The alternative would be to modify snapcraft.yaml so that for instance if a VERSION file is not present, then instead of using a default of 0.0.0, we first try to check for a git tag and use that, i.e. use something like
`
/usr/bin/git describe --tags --abbrev=0 | sed "s/v1/1/"
`
If that fails, then the last resort could be to use 0.0.0.